src/Microsoft.AspNet.Identity/RoleManager.cs

```diff
@@ -162,20 +162,23 @@ public void Dispose()
             {
                 throw new ArgumentNullException("role");
             }
-
             var result = await ValidateRoleInternal(role);
             if (!result.Succeeded)
             {
                 return result;
             }
             await UpdateNormalizedRoleNameAsync(role);
-            return await LogResultAsync(await Store.CreateAsync(role, CancellationToken), role);
+            result = await Store.CreateAsync(role, CancellationToken);
+            using (IdentityLogger.BeginScope(await GetScopeMessageforRoleAsync(role)))
+            {
+                return IdentityLogger.Log(result);
```

Comment:
BTW, I know we are doing it this way consistently because we only identify the user in the scope message. But still wondering if single entry scopes are supposed to be.